codegen/masm/src/masm/program.rs

```diff
@@ -162,14 +164,8 @@ impl Program {
                 masm::LibraryPath::new(entry_import.name.as_str()).expect("invalid module path");
             let entry_id =
                 masm::ProcedureId::from_name(entry.function.as_str(), &entry_module_path);
-            let entry_name = masm::ProcedureName::try_from(
-                FunctionIdent {
-                    module: Ident::with_empty_span(entry_import.alias),
-                    ..entry
-                }
-                .to_string(),
-            )
-            .expect("invalid entrypoint function name");
+            let entry_name = masm::ProcedureName::try_from(entry.function.as_str())
```

Comment:
**Question:** I'm not sure how this could work - we can't reference functions (other than locals) without the parent module name. Specifically the name associated with a module when it is `use`'d. For example, functions imported via `use intrinsics::i32` must be called like `i32::icmp`; while functions imported via `use intrinsics::i32->int32` would need to be called as `int32::icmp`, we can't just call `icmp` anywhere unless we're literally in the `intrinsics::i32` module.

Reply:
I believe I made this change because `ProcedureName::try_from` does not accept `::` in the name. I was not sure if it was a bug so I made this workaround but forgot to document it.

Reply:
Ah, I think it's the `ProcedureId` which is qualified with the module, the `ProcedureName` is unqualified. I think you're right about the change here, I'll have to confirm whether we're emitting the correct `ProcedureId` though.